lang/python/xtdb/test/test_encoding.py

```diff
@@ -107,3 +107,29 @@ def test_timestamps(client):
     client.submit_tx([SqlTx("INSERT INTO trades (xt$id, price) VALUES (1, 100)")])
     query_result = client.query(Sql("SELECT trades.xt$id, trades.price, trades.xt$system_from, trades.xt$system_to FROM trades FOR ALL SYSTEM_TIME"))
     assert isinstance(query_result[0]["xt$system_from"], datetime)
+
+def test_basis_change(client):
+    client.set_basis("2020-01-01T12:34:56Z")
```

Comment:
basis only relates to querying, these should be `set_tx_time` instead?

Reply:
The basis for the client changes both the basis for querying and the system time for the transaction. The tests pass so it definitely works. But it's true that it can be made clearer